relayer/message_relayer.go

```diff
@@ -30,16 +28,33 @@ func (r *Relayer) RelayMessages(ctx context.Context, locker sync.Locker) error {
 	return handleProcessError(err)
 }
 
+func (r *Relayer) getMsgOffsetData(ctx context.Context) (int, int, error) {
+	snapshot, err := r.palomaClient.QueryGetSnapshotByID(ctx, 0)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	numValidators := len(snapshot.Validators)
+	for i, validator := range snapshot.Validators {
+		if r.palomaClient.GetValidatorAddress().String() == validator.Address.String() {
+			return numValidators, i, nil
+		}
+	}
+
+	// If we made it here, somehow our validator address wasn't in the snapshot
+	return 0, 0, chain.ErrNotValidator
+}
+
 func (r *Relayer) relayMessages(ctx context.Context, processors []chain.Processor) error {
 	if len(processors) == 0 {
 		return nil
 	}
 
-	ctx, cleanup, err := collision.GoStartLane(ctx, r.palomaClient, r.palomaClient.GetValidatorAddress())
+	// Get the offset for this validator so that we know which messages to relay
```

Comment:
Could this risk that some validators NEVER get any messages to relay?

Reply:
There shouldn't be any risk of that.  The msg IDs are always increasing, they never reset.  Each new msg will go to the next validator in the list.  The first validator won't get another message until the last validator has see none.